### DISCUSSION TITLE
"Messages Behind" in kafka topic

### DISCUSSION THREAD

**Discussion by winnielim347** [provectus/kafka-ui#2556] [Q&A] [answered]
May i know what is "Messages Behind" in kafka topic means?

Sometime i get number in this column, sometime not.
Is the desired behavior for "Messages Behind" shall be 0?

thanks in advance.
![kafka_message_behind](https://user-images.githubusercontent.com/112146372/189311180-f0d28432-eb03-4061-a9ba-85a4a8f4516d.jpg)

**Comment by Haarolean** [comment, marked as answer]
Hi,

it's also known as consumer lag. Yea, ideally it should be zero.

**Reply by winnielim347** [reply]
Hi @Haarolean 
1.May i know normally what will cause consumer lag?
2.Is "Messages Behind" also means those messages haven't read by consumer yet?

**Reply by Haarolean** [reply]
1. Your consumers consume messages slower than producers produce them.
2. Yes, that's right.

**Comment by xiaomlove** [comment]
I've got a negative number here. What does that mean?

**Reply by Devarsh23** [reply]
@xiaomlove Were you able to figure out why you were seeing the negative number?
We are seeing the same negative number for some kafka-connect based consumer

**Reply by ghost** [reply]
Same here. I constantly get negative numbers in a lot of consumers. What does this mean?
![Screenshot 2024-03-14 at 11 00 04](https://github.com/provectus/kafka-ui/assets/102512135/364af2ac-74b5-4578-965a-8bbcdfb6ff71)

**Reply by Haarolean** [reply]
This is considered normal, meaning your consumers are consuming the messages from the topic faster than they're produced into one.

**Comment by AliAhmadi99** [comment]
The problem is the message bus gets stuck because consumers can't consume a message correctly. I want to be able to skip messages if a error happened in running
